src/vm_controller.cpp

```diff
@@ -125,7 +125,11 @@ std::string vm_controller::generate_command(const jobT &job, size_t /*counter*/,
 	// remove last ','
 	host_list.pop_back();
 
-	return "mpiexec -np " + std::to_string(job.nprocs) + " -hosts " + host_list + " " + job.command;
+	std::string command = job.generate_call();
+	command.replace(command.find("PONCOS_HOSTS"), std::string("PONCOS_HOSTS").size(), host_list);
```

Comment:
I don't think that is a good way to generalize the creating of the mpiexec command. If we just need to add the hosts than that should be a parameter to the generate_call(). But this only makes sense if it also supports the cgroups beackend. Otherwise the code is controller specific and it should be here completely.